shuup/default_tax/models.py

```diff
@@ -86,6 +87,8 @@ def matches(self, taxing_context):
         return True
 
     def save(self, *args, **kwargs):
+        self._postal_codes_min = None
+        self._postal_codes_max = None
         if self.postal_codes_pattern:
             min_value, max_value = Pattern(self.postal_codes_pattern).get_alphabetical_limits()
             self._postal_codes_min = min_value
```

Comment:
You may just remove the `if self.postal_codes_pattern:` line, since `_postal_codes_min` and `_postal_codes_max` should always be set by and `get_alphabetical_limits` should always return the correct value for them.